app/src/main/java/org/p2p/wallet/root/RootPresenter.kt

```diff
@@ -27,14 +39,44 @@ class RootPresenter(
     private fun loadInitialData() {
         launch {
             try {
-                awaitAll(
-                    async { sellInteractor.loadSellAvailability() },
-                    async { orcaInfoInteractor.load() },
-                    async { feeRelayerAccountInteractor.getRelayInfo() }
-                )
+                getInitialDataRequests().awaitAll()
             } catch (e: Throwable) {
-                Timber.e(e, "Error loading initial data")
+                Timber.tag(TAG).e(e, "Error loading initial data")
+            }
+        }
+    }
+
+    private fun getInitialDataRequests(): List<Deferred<Any>> = buildList {
+        add(async { orcaInfoInteractor.load() })
+        add(async { feeRelayerAccountInteractor.getRelayInfo() })
+
+        addIfToggleEnabled(
+            toggle = sellEnabledFeatureToggle,
+            request = sellInteractor::loadSellAvailability,
+        )
+        addIfToggleEnabled(
+            toggle = newSwapEnabled,
+            request = swapTokensRepository::getTokens,
+        )
+        addIfToggleEnabled(
+            toggle = newSwapEnabled,
+            request = swapRoutesRepository::loadAllSwapRoutes,
+        )
+    }
+
+    private fun MutableList<Deferred<Any>>.addIfToggleEnabled(
```

Comment:
redundant wrapping in runCatching, async will not throw error before you will not call awaitAll(). And you can move async from this function. And wrap it in method above.